cmd/osbuild-worker/jobimpl-osbuild.go

```diff
@@ -70,17 +70,50 @@ func osbuildStagesToRPMs(stages []osbuild.StageResult) []koji.RPM {
 }
 
 func (impl *OSBuildJobImpl) Run(job worker.Job) error {
-	outputDirectory, err := ioutil.TempDir(impl.Output, job.Id().String()+"-*")
-	if err != nil {
-		return fmt.Errorf("error creating temporary output directory: %v", err)
+	// Initialize variables needed for reporting back to osbuild-composer
+	var outputDirectory string
+	var r []error
+	var targetResults []*target.TargetResult
+	var osbuildOutput *osbuild.Result = &osbuild.Result{
+		Success: false,
 	}
+
 	defer func() {
-		err := os.RemoveAll(outputDirectory)
+		var targetErrors []string
+		for _, err := range r {
+			errStr := err.Error()
+			fmt.Printf("target errored: %s", errStr)
+			targetErrors = append(targetErrors, errStr)
+		}
+
+		var uploadstatus string = "failure"
+		if len(targetErrors) == 0 {
+			uploadstatus = "success"
+		}
+
+		// In all cases it is necessary to report result back to osbuild-composer worker API
+		err := job.Update(&worker.OSBuildJobResult{
+			Success:       osbuildOutput.Success && len(targetErrors) == 0,
```

Comment:
This seems fragile, could we instead set a variable `success := true` at the end of the `Run()` function and rely on that? That would require us to exit early on errors.

Reply:
Hopefully fixed in following commits.